spark/build.gradle

```diff
@@ -18,16 +18,19 @@
  */
 
 // add enabled Spark version modules to the build
-def sparkVersions = (System.getProperty("sparkVersions") != null ? System.getProperty("sparkVersions") : System.getProperty("defaultSparkVersions")).split(",")
+List<String> sparkVersions = System.getProperty("sparkVersions", System.getProperty("defaultSparkVersions")).split(",").findAll { !it.empty }
+List<String> knownSparkVersions = System.getProperty("knownSparkVersions").split(",")
 
-if (sparkVersions.contains("3.3")) {
-  apply from: file("$projectDir/v3.3/build.gradle")
+for (ver in sparkVersions) {
+  apply from: file("$projectDir/v${ver}/build.gradle")
 }
 
-if (sparkVersions.contains("3.4")) {
-  apply from: file("$projectDir/v3.4/build.gradle")
-}
-
-if (sparkVersions.contains("3.5")) {
-  apply from: file("$projectDir/v3.5/build.gradle")
+tasks.named("clean", Delete) {
+  for (ver in knownSparkVersions) {
+    delete("$projectDir/v${ver}/spark/build")
```

Comment:
this (and other `build`) folder(s) seem to be properly cleaned up when running `./gradlew clean` so why would we want to add them here?

Reply:
"versions" that are not enabled in the current build are _not_ properly cleaned up, right?

Reply:
one could argue that `clean` should only be running against the Spark version(s) that are currently being passed to Gradle rather than calling clean against all known Spark modules